### ISSUE TITLE
[Macros] extract value of a client-write

### ISSUE THREAD

**Issue by hexpwn** [NordicSemiconductor/Android-nRF-Connect#199]
Currently is macros it is possible to `wait-for-write`; but other than asserting the value that was received, we cannot use that value elsewhere.

One example would be sending a notification that incorporates that value (maybe for a request-response situation, for example). 

It would be nice if we could save the received value to a variable of sorts and use it elsewhere in other operations.

**Comment by philips77** [comment]
Yes, macros are very limited in that sense. After all, they are just macros that can be recorded and played back. There is no branching or variables, unfortunately. We will think of how we can improve that. Thank you for your suggestion.
